Src/Base/AMReX_Math.H

```diff
@@ -130,7 +130,7 @@ std::pair<double,double> sincos (double x)
     r.first = sycl::sincos(x, &r.second);
 #elif defined(__CUDA_ARCH__) && defined(AMREX_USE_CUDA) || \
       defined(__HIP_DEVICE_COMPILE__) && defined(AMREX_USE_HIP) || \
-      defined(_GNU_SOURCE)
+      (defined(_GNU_SOURCE) && !defined(__apple_build_version__))
```

Comment:
Ooops, also with vanilla (conda-forge) clang on macos.

We need to use `__APPLE__` or `__OSX__` here.